pkg/aws/ec2/api/cleanup/eni_cleanup.go

```diff
@@ -0,0 +1,221 @@
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cleanup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/aws/ec2/api"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	rcHealthz "github.com/aws/amazon-vpc-resource-controller-k8s/pkg/healthz"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/utils"
+
+	ec2Errors "github.com/aws/amazon-vpc-resource-controller-k8s/pkg/aws/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/go-logr/logr"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+)
+
+// NetworkInterfaceManager interface allows to define the ENI filters and checks if ENI should be deleted for different callers like in the periodic cleanup routine or
+// during node termination
+type NetworkInterfaceManager interface {
+	GetENITagFilters() []*ec2.Filter
+	ShouldDeleteENI(eniID *string) bool
+	UpdateAvailableENIsIfNeeded(eniMap *map[string]struct{})
+	UpdateCleanupMetrics(vpcrcAvailableCount int, vpccniAvailableCount int, leakedENICount int)
+}
+
+type ENICleaner struct {
+	EC2Wrapper api.EC2Wrapper
+	Manager    NetworkInterfaceManager
+	VpcId      string
+	Log        logr.Logger
+}
+
+// common filters for describing network interfaces
+var CommonNetworkInterfaceFilters = []*ec2.Filter{
+	{
+		Name:   aws.String("status"),
+		Values: []*string{aws.String(ec2.NetworkInterfaceStatusAvailable)},
+	},
+	{
+		Name: aws.String("tag:" + config.NetworkInterfaceOwnerTagKey),
+		Values: aws.StringSlice([]string{config.NetworkInterfaceOwnerTagValue,
+			config.NetworkInterfaceOwnerVPCCNITagValue}),
+	},
+}
+
+// ClusterENICleaner periodically deletes leaked network interfaces(provisioned by the controller or VPC-CNI) in the cluster
+type ClusterENICleaner struct {
+	ClusterName   string
+	shutdown      bool
+	ctx           context.Context
+	availableENIs map[string]struct{}
+	*ENICleaner
+}
+
+func (e *ClusterENICleaner) SetupWithManager(ctx context.Context, mgr ctrl.Manager, healthzHandler *rcHealthz.HealthzHandler) error {
+	e.ctx = ctx
+	e.availableENIs = make(map[string]struct{})
+	healthzHandler.AddControllersHealthCheckers(
+		map[string]healthz.Checker{
+			"health-interface-cleaner": rcHealthz.SimplePing("interface cleanup", e.Log),
+		},
+	)
+
+	return mgr.Add(e)
+}
+
+// StartENICleaner starts the ENI Cleaner routine that cleans up dangling ENIs created by the controller
+func (e *ClusterENICleaner) Start(ctx context.Context) error {
+	e.Log.Info("starting eni clean up routine")
+
+	// Start routine to listen for shut down signal, on receiving the signal it set shutdown to true
```

Comment:
don't think we need this comments which just describes code.

Reply:
Makes sense, this was just copied over from the existing eni_cleanup file.